mteb/evaluation/evaluators/RerankingEvaluator.py

```diff
@@ -129,16 +137,37 @@ def compute_metrics_batched(self, model):
             if num_pos == 0 or num_neg == 0:
                 continue
 
+            for doc in instance["positive"]:
+                qrels[query_id][doc] = 1
```

Comment:
It would be best to grab the real qrels here, some of these have different scores (e.g. ClinicalTrials having 1 and 2, which are used for nDCG calculations).

Reply:
Hmm, Good Point , I think then it would be better to keep the dataset in a retrieval format, rather than changing it to reranker format, this was one thing I was also concerned about.
![Screenshot 2024-06-01 at 1 30 35 AM](https://github.com/embeddings-benchmark/mteb/assets/53102161/9dfb1c8b-cc74-47b8-a25f-54d3c3a97fe5)

Reply:
Ah I see.  I guess it depends on if @DPWXY used the qrels as is, or if they binarized the qrels to "relevant" and "non-relevant".  

It will be hard to migrate to `retrieval` because there the pool of documents is combined, whereas this is reranking where each pool is separate.

Reply:
We use the real qrels for nCDG calculation. Especially for the Doris-Mae dataset, the qrels are continuous scores rather than binarized "relevant" and "non-relevant". And ClinicalTrials having 0, 1 and 2 relevance scores for nDCG calculations.

Reply:
@bp-high why don’t we add qrels as an optional data input to the abstract task? I’m surprised it didn’t have nDCG already, probably because of this issue.

So check for the existence of a dataset split called qrels and if it exists, load it.